### PR TITLE
bpo-33800: Fix default argument for parameter dict_type of ConfigParser/RawConfigParser

### PR DESCRIPTION
**Pull request by andresdelfino**
https://bugs.python.org/issue33800

### CODE REVIEW

File: Doc/library/configparser.rst [hunk 1]
```diff
@@ -901,9 +901,6 @@ ConfigParser Objects
    converter gets its own corresponding :meth:`get*()` method on the parser
    object and section proxies.
 
-   .. versionchanged:: 3.1
-      The default *dict_type* is :class:`collections.OrderedDict`.
-
```
Comment (merwok) [inline Doc/library/configparser.rst]: Please do not remove historical notes.

File: Doc/library/configparser.rst [hunk 2]
```diff
@@ -917,6 +914,8 @@ ConfigParser Objects
       providing consistent behavior across the parser: non-string
       keys and values are implicitly converted to strings.
 
+   .. versionchanged:: 3.7
+      The default *dict_type* is :class:`dict`.
```
Comment (merwok) [inline Doc/library/configparser.rst]: Suggestion to avoid puzzlement: ```:class:`dict`, which is now ordered.```

**Comment by andresdelfino** [comment]
I agree!

Thanks for the review Éric :)

File: Doc/library/configparser.rst [hunk 3]
```diff
@@ -915,7 +918,8 @@ ConfigParser Objects
       keys and values are implicitly converted to strings.
 
    .. versionchanged:: 3.7
-      The default *dict_type* is :class:`dict`.
+      The default *dict_type* is :class:`dict`, since it now guarantees
+      insertion order.
```
Comment (merwok) [inline Doc/library/configparser.rst]: This sounds a little funny to me.  What do you think of `preserves` or `keeps` rather than `guarantees`?

Comment (andresdelfino) [reply Doc/library/configparser.rst]: I like preserves too :)

File: Doc/library/configparser.rst [hunk 4]
```diff
@@ -1088,10 +1091,10 @@ ConfigParser Objects
       given *section*.  Optional arguments have the same meaning as for the
       :meth:`get` method.
 
-   .. versionchanged:: 3.8
-      Items present in *vars* no longer appear in the result.  The previous
-      behaviour mixed actual parser options with variables provided for
-      interpolation.
+      .. versionchanged:: 3.8
```
Comment (ned-deily) [inline Doc/library/configparser.rst]: It looks like you are addressing another problem here, one that is only in master.  With this change, it won't be possible to auto backport this fix to 3.7.  Suggest you include this in another PR if necessary.

**Review by ned-deily — CHANGES_REQUESTED** [review]

File: Doc/library/configparser.rst (hunk 4 above)
Comment (andresdelfino) [reply Doc/library/configparser.rst]: Yes, exactly. I'll update the PR.

**Comment by andresdelfino** [comment]
I have made the requested changes; please review again

**Review by ned-deily — APPROVED** [review]

**Pull request merged**